src/Akka.Hosting/AkkaConfigurationBuilder.cs

```diff
@@ -190,47 +191,55 @@ public AkkaConfigurationBuilder StartActors(ActorStarter starter)
         }
 
         public AkkaConfigurationBuilder WithCustomSerializer(
-            string serializerIdentifier, IEnumerable<Type> boundTypes, Func<ExtendedActorSystem, Serializer> serializerFactory)
+            string serializerIdentifier, IEnumerable<Type> boundTypes,
+            Func<ExtendedActorSystem, Serializer> serializerFactory)
         {
             var serializerRegistration = new SerializerRegistration(serializerIdentifier,
                 boundTypes.ToImmutableHashSet(), serializerFactory);
-            _serializers.Add(serializerRegistration);
+            Serializers.Add(serializerRegistration);
             return this;
         }
 
-        internal void Build()
+        internal void Bind()
         {
-            if (!_complete)
-            {
-                _complete = true;
-                _serviceCollection.AddSingleton<AkkaConfigurationBuilder>(this);
+            // register as singleton - not interested in supporting multi-Sys use cases
+            ServiceCollection.AddSingleton<ActorSystem>(ActorSystemFactory());
 
-                // start the IHostedService which will run Akka.NET
-                _serviceCollection.AddHostedService<AkkaHostedService>();
+            ServiceCollection.AddSingleton<ActorRegistry>(sp =>
+            {
+                return ActorRegistry.For(sp.GetRequiredService<ActorSystem>());
+            });
+        }
 
+        private  static Func<IServiceProvider, ActorSystem> ActorSystemFactory()
```

Comment:
The `ActorSystem` construction now uses the DI container to resolve the `AkkaConfigurationBuilder` in order to keep context boundaries clean. As a result most of the `private` members of the `AkkaConfigurationBuilder` are now `internal` so they can be accessed from this delegate.